torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -237,9 +241,77 @@ def __init__(self, tolerance=0.5):
         self.tolerance = tolerance
         self.useful_observer_fqns: Set[str] = set([])
 
-    def determine_observer_insert_points(self, model) -> Tuple[Set[str], Any]:
+    def determine_observer_insert_points(self, prepared_fx_model: GraphModule) -> Dict[str, Dict[str, Any]]:
```

Comment:
I think a cleaner interface here would be the detector will just insert the observers directly, rather than outputting locations for `ModelReport` to insert them later. This way you don't have to return another dict with new keys and values, which adds complexity to the code. Then `ModelReport` can just call `detector.insert_observers` one by one for each detector it has.

Reply:
If you're worried about duplicate code you can make `ModelReport.__insert_observer_around_module` a helper function not related to the class. Then each detector can call this helper function when inserting observers.

Reply:
the worry is that multiple detectors will need to insert observers into the same place, @vspenubarthi 's milestone 2 is to do this for equalization so by doing it this way, we can prevent redundancy.

I felt like this seemed a little cleaner than having every observer insertion having their own method and each of those having to check for existing observers in order to prevent redundancy..etc, let me know if you disagree though @andrewor14

Reply:
I see, got it. That makes sense